ONSdigital/ras-collection-instrument
# Upgrade sqlalchemy 2.0

# What and why?

sqlalchemy 2.0 finally released so we upgraded to it to keep current.
It's not required too many changes as we used sessions and ORM models in a fairly simple way.

# How to test?

- Deploy app to environment
- Run all 4 types of acceptance tests to ensure it works
- Try a few manual things (deleting instruments, etc)

To test it with a completely fresh database
- delete the ras_ci schema
- `kubectl rollout restart deployment collection-instrument`
- run the acceptance tests again to verify that it's happy

This will likely have a few errors in the logs as all the workers will try and run alembic, and a few might fail initially because of race conditions but it all works out in the end.

# Trello
